x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -4,90 +4,193 @@
-import { LogsFetchDataResponse } from '../../../observability/public';
+import { InfraClientCoreSetup, InfraClientStartDeps } from '../types';
+import {
+  FetchData,
+  LogsFetchDataResponse,
+  HasData,
+  FetchDataParams,
+} from '../../../observability/public';
 
-export function getLogsHasDataFetcher(getStartServices: InfraClientCoreSetup['getStartServices']) {
-  return async () => {
-    // if you need the data plugin, this is how you get it
-    // const [, startPlugins] = await getStartServices();
-    // const { data } = startPlugins;
+interface StatsAggregation {
+  buckets: Array<{ key: string; doc_count: number }>;
+}
 
-    // if you need a core dep, we need to pass in more than just getStartServices
+interface SeriesAggregation {
+  buckets: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+    dataset: StatsAggregation;
+  }>;
+}
+
+interface LogParams {
+  index: string;
+  timestampField: string;
+}
+
+type StatsAndSeries = Pick<LogsFetchDataResponse, 'stats' | 'series'>;
+
+export function getLogsHasDataFetcher(
+  getStartServices: InfraClientCoreSetup['getStartServices']
+): HasData {
+  return async () => {
+    const [, startPlugins] = await getStartServices();
+    const { data } = startPlugins;
 
-    // perform query
-    return true;
+    return await hasLogsOverview('filebeat-*', data);
   };
 }
 
 export function getLogsOverviewDataFetcher(
   getStartServices: InfraClientCoreSetup['getStartServices']
-) {
-  return async (): Promise<LogsFetchDataResponse> => {
-    // if you need the data plugin, this is how you get it
-    // const [, startPlugins] = await getStartServices();
-    // const { data } = startPlugins;
+): FetchData<LogsFetchDataResponse> {
+  return async (params) => {
+    const [, startPlugins] = await getStartServices();
+    const { data } = startPlugins;
 
-    // if you need a core dep, we need to pass in more than just getStartServices
+    // FIXME figure out how to get these from the sourceConfiguration
+    const { stats, series } = await fetchLogsOverview(
+      { index: 'filebeat-*', timestampField: '@timestamp' },
+      params,
+      data
+    );
 
-    // perform query
     return {
       title: 'Log rate',
       appLink: 'TBD', // TODO: what format should this be in, relative I assume?
-      stats: {
-        nginx: {
-          type: 'number',
-          label: 'nginx',
-          value: 345341,
-        },
-        'elasticsearch.audit': {
-          type: 'number',
-          label: 'elasticsearch.audit',
-          value: 164929,
+      stats,
+      series,
+    };
+  };
+}
+
+async function hasLogsOverview(
+  index: string,
+  dataPlugin: InfraClientStartDeps['data']
+): Promise<boolean> {
+  const esSearcher = dataPlugin.search.getSearchStrategy('es');
+  return new Promise((resolve, reject) => {
+    esSearcher
+      .search({
+        params: {
+          index,
+          body: {
+            size: 0,
```

Comment:
You could also add `terminateAfter: 1`

Reply:
Ah! I didn't know about that.

I changed the approach anyway, calling an API that returns if any of the indices exists instead of checking if they have data.